optuna/integration/mlflow.py

```diff
@@ -0,0 +1,97 @@
+import mlflow
+
+import optuna
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Dict  # NOQA
+
+
+class MlflowCallback(object):
+    """Callback to track optuna trials with Mlflow.
+
+    This callback adds all relevant information that is
+    tracked by Optuna to Mlflow.
+
+    Example:
+
+        Add Mlflow callback to optuna optimization.
+
+        .. code::
```

Comment:
fixed

Reply:
I am back to `code` instead of `testcode` because  I can not provide a valid `tracking_uri`. I have not http URL and do not want to write to a directory. What do you think?

Reply:
unresolving again

Reply:
@PhilipMay 
Regarding [the doctest problem](https://github.com/optuna/optuna/pull/1028#discussion_r396016459), how about using `tempfile.mkdtemp` and cleaning up after testing? For example:

```python
        .. testsetup::
            import pathlib
            import tempfile
            tempdir = tempfile.mkdtemp()
            YOUR_TRACKING_URI = pathlib.Path(tempdir).as_uri()
        .. testcode::
            import optuna
            from optuna.integration.mlflow import MlflowCallback

            def objective(trial):
                x = trial.suggest_uniform('x', -10, 10)
                return (x - 2) ** 2

            mlflc = MlflowCallback(
                tracking_uri=YOUR_TRACKING_URI,
                metric_name='my metric score',
            )

            study = optuna.create_study(study_name='my_study')
            study.optimize(objective, n_trials=10, callbacks=[mlflc])
        .. testcleanup::
            import shutil
            shutil.rmtree(tempdir)
```

With `testsetup` and `testcleanup`, you can keep the code readability by hiding the setup and cleanup procedures. The picture below is generated docs.

![image](https://user-images.githubusercontent.com/1061922/77241539-bd688980-6c36-11ea-8e5f-bd5180e3e124.png)

Reply:
many thanks - I made the change